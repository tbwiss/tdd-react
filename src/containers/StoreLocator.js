import React from 'react';
import Header from '../components/Header';
import Button from '../components/Button';
import Map from '../components/Map';
import mapChooser from '../mapChooser';
import axios from 'axios';

class StoreLocator extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currentMap: 'none.png',
      shops: []
    }
    this.chooseMap = this.chooseMap.bind(this);
  }

  async componentDidMount() {
    const response = await axios.get('http://localhost:3000/data/shops.json');
    this.setState({ shops: response.data.shops });
  }

  chooseMap(event) {
    if (event && event.persist) {
      event.persist();
    }
    this.setState({ currentMap: mapChooser(event.target.value) });
  }

  render () {
    return (
      <>
        <Header/>
        <div>
          {
            this.state.shops.map(shop => (
              <Button
                handleClick={event => this.chooseMap(event)}
                key={`${shop.location}+${shop.address}`}
                location={shop.location}
              />
            ))
          }
        </div>
        <Map 
          imageName={this.state.currentMap}
        />
      </>
    );
  }
}

export default StoreLocator;