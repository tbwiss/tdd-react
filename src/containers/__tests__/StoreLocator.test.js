import React from 'react';
import { shallow } from 'enzyme';
import StoreLocator from '../StoreLocator';

describe('StoreLocator', () => {
  let mountedStoreLocator;

  beforeEach(() => {
    mountedStoreLocator = shallow(<StoreLocator />);
  });

  it('renders without crashing', () => {
    shallow(<StoreLocator />);
  });

  it('renders a header', () => {
    const headers = mountedStoreLocator.find('Header');
    expect(headers.length).toBe(1);
  });

  it('renders two buttons', () => {
    const buttons = mountedStoreLocator.find('Button');
    expect(buttons.length).toBe(3);
  });

  it('renders a map', () => {
    const maps = mountedStoreLocator.find('Map');
    expect(maps.length).toBe(1);
  });
});

describe('chooseMap', () => {
  it('update state currentMap using the location passed to it', () => {
    let mountedStoreLocator = shallow(<StoreLocator />);
    let mockEvent = { target: { value: 'testland' } };
    mountedStoreLocator.instance().chooseMap(mockEvent);
    expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png');
  });
});
