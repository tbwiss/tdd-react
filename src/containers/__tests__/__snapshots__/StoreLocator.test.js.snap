// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocator renders correctly 1`] = `
Array [
  <div
    className="Header"
  >
    <img
      alt="Wired Brain logo"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>,
  <div
    className="buttons"
  />,
  <div
    className="MapBox"
  >
    <img
      alt="the map"
      src="images/none.png"
    />
  </div>,
]
`;
